src/tools/compiletest/src/header.rs

```diff
@@ -982,3 +953,34 @@ fn parse_normalization_string(line: &mut &str) -> Option<String> {
     *line = &line[end + 1..];
     Some(result)
 }
+
+// Takes a directive of the form "<version1> [- <version2>]",
+// returns the numeric representation of <version1> and <version2> as
+// tuple: (<version1> as u32, <version2> as u32)
+// If the <version2> part is omitted, the second component of the tuple
+// is the same as <version1>.
+fn extract_version_range<F>(line: &str, parse: F) -> (u32, u32)
+where
+    F: Fn(&str) -> Option<u32>,
+{
+    let range_components = line
+        .split(&[' ', '-'][..])
+        .filter(|word| !word.is_empty())
+        .map(parse)
+        .skip_while(Option::is_none)
+        .take(3) // 3 or more = invalid, so take at most 3.
+        .collect::<Vec<Option<u32>>>();
+
+    match *range_components {
+        [v] => {
+            let v = v.unwrap();
+            (v, v)
+        }
+        [min, max] => {
+            let v_min = min.unwrap();
+            let v_max = max.expect("Malformed version directive");
+            (v_min, v_max)
+        }
+        _ => panic!("Malformed version directive"),
```

Comment:
Both of these will be incredibly hard to track down (i.e., what version range is malformed) -- could we at least print out the line we're attempting to parse here? It might be easier to have this return a Option and do an unwrap_or_else at the call sites or something like that.

Reply:
I pushed a new commit. But I don't like it much.